app/grandchallenge/discussion_forums/management/commands/migrate_topic_tracks.py

```diff
@@ -0,0 +1,80 @@
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.management import BaseCommand
+from machina.apps.forum.models import Forum as MachinaForum
+from machina.apps.forum_conversation.models import Topic as MachinaTopic
+from machina.apps.forum_tracking.models import TopicReadTrack
+
+from grandchallenge.discussion_forums.models import (
+    ForumTopic,
+    TopicReadRecord,
+    get_matching_topic,
+)
+
+
+class Command(BaseCommand):
+    help = "Migrates topic read records"
+
+    def handle(self, *args, **options):
+        batch_size = 1000
+        batch = []
+        total_created = 0
+        num_tracks_to_create = (
+            TopicReadTrack.objects.filter(migrated_track__isnull=True)
+            .exclude(user__username=settings.ANONYMOUS_USER_NAME)
+            .count()
+        )
+
+        if num_tracks_to_create == 0:
+            self.stdout.write(
+                self.style.SUCCESS("No TopicReadTracks to migrate")
+            )
+            return
+
+        for track in (
+            TopicReadTrack.objects.filter(migrated_track__isnull=True)
+            .exclude(user__username=settings.ANONYMOUS_USER_NAME)
+            .iterator(chunk_size=1000)
+        ):
+            try:
+                new_topic = get_matching_topic(
+                    old_topic_id=track.topic.pk,
+                    old_topic_model=MachinaTopic,
+                    new_topic_model=ForumTopic,
+                    old_forum_model=MachinaForum,
+                )
+            except ObjectDoesNotExist:
+                continue
+
+            batch.append(
+                TopicReadRecord(
+                    source_object=track,
+                    topic=new_topic,
+                    user=track.user,
+                    created=track.mark_time,  # the original model does not have a creation time stamp
+                    modified=track.mark_time,
+                )
+            )
```

Comment:
Currently any user on the platform can view any post in any forum. In the new implementation only challenge participants and admins can read their forum. I think that makes sense. However, when it comes to migrating the `TopicReadTracks`, it means we might end up adding inconsistent data if we just blindly copy over (non-participant read record for a forum topic). 

So I will need to add a check here that `track.user` has view permission for the `new_topic` before creating the `TopicReadRecord`. We already have this check in place on the new views/forms (see recent PR), but I had forgotten to add this in the migration.

Reply:
I have added the permission check